### PR TITLE
object/ks3: use virtual hosted-style for ks3

### LINKED ISSUE #2348: juicefs format error when using new bucket in ks3
**What happened**:
```
juicefs format --storage=ks3 --bucket=https://minimax-test-reources.ks3-cn-beijing-internal.ksyuncs.com --access-key=***** --secret-key=******  redis://192.168.1.90/3 test
2022/07/11 07:08:58.105862 juicefs[720] <INFO>: Meta address: redis://192.168.1.90/3 [interface.go:400]
2022/07/11 07:08:58.107594 juicefs[720] <WARNING>: maxmemory_policy is "volatile-lru",  we will try to reconfigure it to 'noeviction'. [info.go:88]
2022/07/11 07:08:58.107671 juicefs[720] <WARNING>: AOF is not enabled, you may lose data if Redis is not shutdown properly. [info.go:83]
2022/07/11 07:08:58.108112 juicefs[720] <WARNING>: reconfigured maxmemory-policy to 'noeviction', but it's still volatile-lru [redis.go:2864]
2022/07/11 07:08:58.108393 juicefs[720] <INFO>: Ping redis: 229.573µs [redis.go:2871]
2022/07/11 07:08:58.108791 juicefs[720] <INFO>: Data use ks3://minimax-test-reources/test/ [format.go:437]
2022/07/11 07:09:10.549455 juicefs[720] <FATAL>: Storage ks3://minimax-test-reources/test/ is not configured correctly: Failed to put: SecondLevelDomainForbidden: The bucket you are attempting to access must be addressed using KS3 third level domain.
 status code: 403, request id: [f4kgut80n5pobs6l858qrnmvlkipdfs3] [format.go:440]
```

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**

**Environment**:
- JuiceFS version (use `juicefs --version`) or Hadoop Java SDK version: v1.0.0-rc2
- Cloud provider or hardware configuration running JuiceFS:
- OS (e.g `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Object storage (cloud provider and region, or self maintained): ks3
- Metadata engine info (version, cloud provider managed or self maintained): redis
- Network connectivity (JuiceFS to metadata engine, JuiceFS to object storage):
- Others:

### PR DESCRIPTION
**Pull request by zhijian-pro**
Close #2348

### CODE REVIEW

File: pkg/object/ks3.go [hunk 1]
```diff
@@ -280,7 +280,7 @@ func newKS3(endpoint, accessKey, secretKey, token string) (ObjectStorage, error)
 		Endpoint:         strings.SplitN(uri.Host, ".", 2)[1],
 		DisableSSL:       !ssl,
 		HTTPClient:       httpClient,
-		S3ForcePathStyle: true,
+		S3ForcePathStyle: false,
```
Comment (davies) [inline pkg/object/ks3.go:283]: We still need this to be `true` for some existing KS3 clusters.

Comment (zhijian-pro) [reply pkg/object/ks3.go:283]: Existing clusters are supported by both modes.

Comment (davies) [reply pkg/object/ks3.go:283]: No, some clusters on premise require S3ForcePathStyle to be true.

**Comment by zhijian-pro** [comment]
Implemented in another [PR](https://github.com/juicedata/juicefs/pull/2353)

**Review by davies — APPROVED** [review]

File: pkg/object/ks3.go (hunk 1 above)
Comment (davies) [reply pkg/object/ks3.go:283]: Users can add *.xxx.com to enable vhost style for their existing clusters.

**Pull request merged**